### ISSUE TITLE
How to use this in my plugin of OctoberCMS

### ISSUE THREAD

**Issue by jacksun101** [spatie/laravel-activitylog#595]
Sorry for the issue of such particular integration with OctoberCMS. I've followed your installation instruction of ActivityLog v2. And I got the following error when I made an activity()->log() call.

> Type error: Argument 1 passed to Spatie\Activitylog\ActivityLogStatus::__construct() must be an instance of Illuminate\Config\Repository, instance of October\Rain\Config\Repository given

And then I modified to use October\Rain\Config\Repository in ActivityLogStatus, another error showed up.

> Target [October\Rain\Config\LoaderInterface] is not instantiable while building [Spatie\Activitylog\ActivityLogStatus, October\Rain\Config\Repository, Spatie\Activitylog\ActivityLogStatus, October\Rain\Config\Repository].

I think maybe I need to create something like a wrapper to make ActivityLog work in OctoberCMS. Can you give me some suggestion or direction? Thanks for such a great library.

**Comment by Gummibeer** [comment]
Could you upgrade to the latest version? We've fixed the config problem in #505 and released it as v3.4.0
This should solve your issue.

**Comment by jacksun101** [comment]
Thanks. But I use v2 for a reason. If I composer require v3.4.0, composer update will cause issue.

> Problem 1
    - Installation request for spatie/laravel-activitylog 3.4.* -> satisfiable by spatie/laravel-activitylog[3.4.0].
    - spatie/laravel-activitylog 3.4.0 requires php ^7.2 -> your PHP version (7.3.7) overridden by "config.platform.php" version (7.0) does not satisfy that requirement.

And "composer install --ignore-platform-reqs" causes more lines of issue.

> Problem 1
    - Conclusion: remove laravel/framework v5.5.48
    - Conclusion: don't install laravel/framework v5.5.48
    - Conclusion: don't install laravel/framework v5.5.47
    - Conclusion: don't install laravel/framework v5.5.46
    ....

If I just apply the #505 fix, then there is another error shows up.

> Unresolvable dependency resolving [Parameter #0 [ <required> $app ]] in class Illuminate\Auth\AuthManager

Is there any easy way that can solve this? Thank you.

**Comment by Gummibeer** [comment]
How have you made Laravel 5.5 run on PHP 7.0? 😲
My recommendation is to upgrade to PHP 7.2 or even better 7.3 and Laravel 5.8 or even better 6.0.
Some links about it:
https://www.php.net/supported-versions.php
https://laravel.com/docs/6.x/releases#support-policy
https://connortumbleson.com/2019/01/28/lts-is-a-lie/

If this is not possible for you: the main problem, that got solved in the PR, is that we required the Laravel config class instead of it's interface.
